go/engine/signup.go

```diff
@@ -212,6 +212,11 @@ func (s *SignupEngine) registerDevice(m libkb.MetaContext, deviceName string) er
 		IsEldest:   true,
 	}
 
+	if !libkb.CheckDeviceName.F(s.arg.DeviceName) {
+		m.CDebugf("invalid device name supplied: %s", s.arg.DeviceName)
+		return fmt.Errorf("Bad device name: %v is not %s", s.arg.DeviceName, libkb.CheckDeviceName.Hint)
```

Comment:
You might consider introducing a new error type in `libkb/errors.go` if you're later going to check that you're getting the right error back.

Reply:
Hmm, `libkb.DeviceBadNameError` is already there, use that...

Reply:
done